tests/test_utils_flock.py

```diff
@@ -0,0 +1,162 @@
+        if exclusive:
+            fd = acquire_write_binary_flock(lock_file=lock_file, logger=mock_logger, timeout_seconds=timeout_seconds)
+        else:
+            fd = acquire_read_binary_flock(lock_file=lock_file, logger=mock_logger, timeout_seconds=timeout_seconds)
+        assert fd is not None
+        time.sleep(sleep_seconds)
+    finally:
+        release_flock(fd, mock_logger)
+
+def test_flock_aquire1(mock_logger):
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+        try:
+            fd = None
+            lock_file = f'{tmp_dir_name}/mylock.file'
+            fd = acquire_write_binary_flock(lock_file=lock_file, logger=mock_logger, timeout_seconds=1.0)
+            assert fd is not None
+
+        finally:
+            if fd:
+                release_flock(fd, mock_logger)
+
+def test_flock_exclusive_aquire(mock_logger):
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+        try:
+            lock_file = f'{tmp_dir_name}/mylock.file'
+            fd = None
+
+            # Test exclusive followed by exclusive acquire
+            p = Process(target=get_and_hold_lock, args=(lock_file, 4))
+            p.start()
+            time.sleep(1)
+            with pytest.raises(FailureToAcquireLockException):
+                fd = acquire_write_binary_flock(lock_file, mock_logger, timeout_seconds=1.0)
+            assert fd is None
+            p.join()
+
+            # Test shared followed by exclusive acquire
+            p = Process(target=get_and_hold_lock, args=(lock_file, 4, False))
+            p.start()
+            time.sleep(1)
+            with pytest.raises(FailureToAcquireLockException):
+                fd = acquire_write_binary_flock(lock_file, mock_logger, timeout_seconds=1.0)
+            assert fd is None
+            p.join()
+
+            # Test exclusive followed by shared acquire
+            p = Process(target=get_and_hold_lock, args=(lock_file, 4, True))
+            p.start()
+            time.sleep(1)
+            with pytest.raises(FailureToAcquireLockException):
+                fd = acquire_read_binary_flock(lock_file, mock_logger, timeout_seconds=1.0)
+            assert fd is None
+            p.join()
+
+        finally:
+            if fd:
+                release_flock(fd, mock_logger)
+
+def test_flock_release(mock_logger):
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+        try:
+            lock_file = f'{tmp_dir_name}/mylock.file'
+            fd = None
+
+            p = Process(target=get_and_hold_lock, args=(lock_file, 2))
+            p.start()
+            time.sleep(1)
+            fd = acquire_write_binary_flock(lock_file, mock_logger, timeout_seconds=2.0)
+            assert fd is not None
+            p.join()
+        finally:
+            if fd:
+                release_flock(fd, mock_logger)
+
+def test_flock_aquire_exlusive_timeout(mock_logger):
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
```

Comment:
Personally, I'd make this tmp_dir_name a fixture to cleanup this test code, but it's fine as is.

Reply:
Yeah I should use fixtures more. But if you are ok with it then I will probably just leave it as is.